packages/@n8n/nodes-langchain/nodes/agents/Agent/agents/ToolsAgent/execute.ts

```diff
@@ -346,11 +346,17 @@ export async function prepareMessages(
 		outputParser?: N8nOutputParser;
 	},
 ): Promise<BaseMessagePromptTemplateLike[]> {
-	const messages: BaseMessagePromptTemplateLike[] = [
-		['system', `{system_message}${options.outputParser ? '\n\n{formatting_instructions}' : ''}`],
-		['placeholder', '{chat_history}'],
-		['human', '{input}'],
-	];
+	const messages: BaseMessagePromptTemplateLike[] =
+		(options.systemMessage ?? ctx.getNode().typeVersion < 1.9)
+			? [
```

Comment:
This would also remove `formatting_instructions` from the system message if output parser is provided but while using the default systemMessage. Might also be worth writing a test for this :P